game-core/src/main/java/games/strategy/triplea/ai/pro/util/MoveBatcher.java

```diff
@@ -129,22 +82,41 @@ private void addMove(final Move newMove) {
    * @param sequences The sequences to try to merge into the sequence.
    */
   private static void mergeSequences(
-      final ArrayList<Move> sequence, final List<ArrayList<Move>> sequences) {
+      final ArrayList<MoveDescription> sequence, final List<ArrayList<MoveDescription>> sequences) {
     for (final var otherSequence : sequences) {
       boolean merge = (otherSequence.size() == sequence.size());
       for (int i = 0; merge && i < sequence.size(); i++) {
-        merge = sequence.get(i).canMergeWith(otherSequence.get(i));
+        merge = canMergeMoves(sequence.get(i), otherSequence.get(i));
       }
       if (!merge) {
         continue;
       }
       for (int i = 0; i < sequence.size(); i++) {
-        if (!sequence.get(i).mergeWith(otherSequence.get(i))) {
-          throw new IllegalStateException(
-              "Could not merge move despite checking canMergeWith() earlier.");
-        }
+        sequence.set(i, mergeMoves(sequence.get(i), otherSequence.get(i)));
       }
       otherSequence.clear();
     }
   }
+
+  private static boolean canMergeMoves(final MoveDescription move1, final MoveDescription move2) {
+    return isTransportLoad(move1) == isTransportLoad(move2)
+        && move1.getRoute().equals(move2.getRoute());
+  }
+
+  private static boolean isTransportLoad(final MoveDescription move) {
+    return !move.getUnitsToTransports().isEmpty();
+  }
+
+  // Merges the two moves. Caller must ensure canMergeMoves() is true before calling.
+  private static MoveDescription mergeMoves(
+      final MoveDescription move1, final MoveDescription move2) {
+    if (!canMergeMoves(move1, move2)) {
```

Comment:
nit, can use guava to assert this in a single statement:

```
Preconditions.checkState(canMergeMoves(move1, move2, "can't merge moves: " + move1 + " and " + move2");
```

Second comment, still nitty, 'illegalArgumentException' could be more appropriate here as we are checking the arguments.